reframe/core/meta.py

```diff
@@ -143,27 +145,34 @@ def __init__(cls, name, bases, namespace, **kwargs):
         # Set up the hooks for the pipeline stages based on the _rfm_attach
```

Comment:
How about moving all this hook logic into the `HookRegistry`'s `__init__`?

Reply:
Perhaps it makes sense. I can give it a try.